pkg/pod/pod.go

```diff
@@ -170,9 +170,11 @@ func (b *Builder) Build(ctx context.Context, taskRun *v1beta1.TaskRun, taskSpec
 		// Mount /tekton/creds with a fresh volume for each Step. It needs to
 		// be world-writeable and empty so creds can be initialized in there. Cant
 		// guarantee what UID container runs with.
```

Comment:
may be useful to update this to mention that it mounts fresh volume for each step _except when creds-init is disabled_